src/convertToObject.js

```diff
@@ -6,7 +6,17 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((str) => str.trim())
+    .filter((str) => str.length !== 0)
+    .reduce((prev, property) => {
+      const [name, value] = property.split(':');
```

Comment:
There is a potential issue here that is not covered by the checklist but is worth mentioning. If there are additional colons in the property values, the split will not work correctly. Although handling multiple colons is not required for this task, it's something to be aware of for more robust code. If you choose to handle it, you could use a regular expression or limit the split to the first colon.